hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -422,11 +423,10 @@ class HoodieSparkSqlWriterInternal {
               genericRecords.map(gr => keyGenerator.getKey(gr).getPartitionPath).toJavaRDD().distinct().collect()
             }
 
-            // Issue the delete.
-            val schemaStr = new TableSchemaResolver(tableMetaClient).getTableAvroSchema.toString
+            val schemaStr = new TableSchemaResolver(tableMetaClient).getTableAvroSchema(false).toString
```

Comment:
Could you create follow-ups on renaming the methods to be legible?  `getTableAvroSchema(false)` -> `getTableAvroSchemaWithoutMetaFields()`, and `getTableAvroSchema()` -> `getTableAvroSchemaWithMetaFields()`, to avoid confusion in the future.

Reply:
Should I create a ticket in Jira for this?

Reply:
In fact it exists(`getTableAvroSchemaWithoutMetadataFields`), will use it instead

Reply:
It's deprecated in favor of one with boolean flag.
```
  /**
   * Gets users data schema for a hoodie table in Avro format.
   *
   * @return  Avro user data schema
   * @throws Exception
   *
   * @deprecated use {@link #getTableAvroSchema(boolean)} instead
   */
```